Numpy argmax 6424 (#6440)
Codecademy/docs: content/numpy/concepts/built-in-functions/terms/argmax/argmax.md

---

> title: argmax()
> description: Returns the indices of the maximum values along a specified axis in an array.
> subjects:
>   - Computer Science
>   - Data Science
> tags:
>   - Arrays
>   - Data
>   - Machine Learning
>   - NumPy
> catalogContent:
>   - learn-python-3
>   - paths/data-science

Reviewer: ```suggestion
Title: '.argmax()'
Description: 'Returns the indices of the maximum values along a specified axis in an array.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Arrays'
  - 'Data'
  - 'Machine Learning'
  - 'NumPy'
CatalogContent:
  - 'learn-python-3'
  - 'paths/data-science'
```

---

> # argmax()
> 

Reviewer: ```suggestion
```

Reviewer: term heading can be removed

> The NumPy `.argmax()` function returns the indices of the maximum values along a specified axis in an array.

Reviewer: ```suggestion
The NumPy **`.argmax()`** function returns the indices of the maximum values along a specified axis in an array.
```

Reviewer: the first instance of the term should be in bold and backticks

## Syntax

> ```python

Reviewer: ```suggestion
```pseudo
```

numpy.argmax(array, axis=None, out=None, keepdims=<no value>)

> 

Reviewer: please describe the parameters here in the following format:

- `array`: its description>
- `axis`: its description>
- `out`: its description>
- `keepdims`: its description>

> import numpy as np

Reviewer: Put this code under the heading:

## Example

and also give a one to two line explanation of the example code

# Create and analyze 1D array
scores = np.array([10, 30, 20, 50, 40])
max_index = np.argmax(scores)
print("Index of max value:", max_index)  # Output: 3
print("Max value:", scores[max_index])   # Output: 50
import numpy as np

# Create 2D array
matrix = np.array([[1, 5, 3], 
                  [4, 2, 6]])

# Analyze columns (axis=0)
print("Max indices per column:", np.argmax(matrix, axis=0))  # Output: [1 0 1]

# Analyze rows (axis=1)
print("Max indices per row:", np.argmax(matrix, axis=1))     # Output: [1 2]

> 

Reviewer: showcase the output of the code in the shell block: ```shell```

```shell
Index of max value: 3
Max value: 50
Max indices per column: [1 0 1]
Max indices per row: [1 2]
```

## Codebyte Example
import numpy as np

# Create a 2D array
data = np.array([[10, 20, 30], [40, 50, 60]])

# Find the index of the maximum value along axis 1 (rows)
result = np.argmax(data, axis=1)
print(result)  # Output: [2 2]

> ![Output example](argmax-output.png)

Reviewer: ```suggestion
```

Reviewer: the output image can be removed from this entry
